src/Spago/Registry.purs

```diff
@@ -21,60 +40,201 @@ import Spago.Git as Git
 import Spago.Paths as Paths
 import Spago.Purs as Purs
 
-type RegistryEnv a =
-  { getManifestFromIndex :: PackageName -> Version -> Spago (LogEnv ()) (Maybe Manifest)
-  , getMetadata :: PackageName -> Spago (LogEnv ()) (Either String Metadata)
-  , offline :: OnlineStatus
+type PreRegistryEnvRow a =
+  ( offline :: OnlineStatus
   , logOptions :: LogOptions
   , purs :: Purs.Purs
   , git :: Git.Git
   , db :: Db.Db
   | a
+  )
+
+type PreRegistryEnv a = Record (PreRegistryEnvRow a)
+
+type RegistryEnvRow a = PreRegistryEnvRow
+  ( getRegistry :: Spago (PreRegistryEnv ()) RegistryFunctions
+  | a
+  )
```

Comment:
I'm not sure I follow why we have the "pre-registry" row and then the "registry" row with the extra `getRegistry` function. Would you mind explaining it a little?

Reply:
Yeah, the compiler would try to infer an infinite type if these would run in the same env, so I had to separate them